### ISSUE TITLE
Question about “Multi-space Alignment”

### ISSUE THREAD

**Issue by Rayn-Wu** [liujiaming1996/BEVUDA#2]
Hi, thanks again for sharing this project.

I would to ask some details about “Multi-space Alignment”.
![image](https://github.com/user-attachments/assets/43f3f92e-19e2-4354-80b8-6219619328ba)

I want to know why we need alignment in BEV space, since we already do the alignment in Voxel space.  I mean if we align the voxel features in target and source domain，isn't the corresponding BEV features already aligned？or am I missing some details of your method?

**Comment by Rayn-Wu** [comment]
BTW，could you please show me where do you get $L_{MKT}$, I looked through this project but didn't find it.
Thanks alot.

**Comment by liujiaming1996** [comment]
> BTW，could you please show me where do you get L M K T , I looked through this project but didn't find it. Thanks alot.

Thank you again for your interest in our work. In the open-source code, we have only provided the BEV feature transfer loss from L(MKT), as shown below. The feature transfer between images and voxels follows a similar approach. Since L(MKT) is fundamentally a classic knowledge distillation loss, exploring newer KD strategies and loss functions could further enhance the accuracy of your student model. In the experiments we conducted two years ago, we only explored using MSE or KD loss as the L(MKT) loss. I hope this explanation supports your research, and I wish you continued success.

self.bev_feature_loss = torch.nn.MSELoss()
or.  self.bev_feature_loss = torch.nn.KLDivLoss()
 bev_feature_loss = self.bev_feature_loss(bev_feats_target,bev_feats_target_student)

**Comment by liujiaming1996** [comment]
> Hi, thanks again for sharing this project.
> 
> I would to ask some details about “Multi-space Alignment”. ![image](https://private-user-images.githubusercontent.com/78060545/374873465-43f3f92e-19e2-4354-80b8-6219619328ba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg3MDg4NDcsIm5iZiI6MTcyODcwODU0NywicGF0aCI6Ii83ODA2MDU0NS8zNzQ4NzM0NjUtNDNmM2Y5MmUtMTllMi00MzU0LTgwYjgtNjIxOTYxOTMyOGJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEyVDA0NDkwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViZTNlNjEyODFjZTVhYmYyMThmYzI2NTE4OTg2ODQxN2VkYjU2OTA2MjU1MTI5MWY0OTc3ZTg0Yzg3MmQwMGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.B1BMh6JlRpraeQ04AIisRb_9W6vk7Yu2MiRhKMDqguY)
> 
> I want to know why we need alignment in BEV space, since we already do the alignment in Voxel space. I mean if we align the voxel features in target and source domain，isn't the corresponding BEV features already aligned？or am I missing some details of your method?

**Comment by liujiaming1996** [comment]
> Hi, thanks again for sharing this project.
> 
> I would to ask some details about “Multi-space Alignment”. ![image](https://private-user-images.githubusercontent.com/78060545/374873465-43f3f92e-19e2-4354-80b8-6219619328ba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg3MDk0MDksIm5iZiI6MTcyODcwOTEwOSwicGF0aCI6Ii83ODA2MDU0NS8zNzQ4NzM0NjUtNDNmM2Y5MmUtMTllMi00MzU0LTgwYjgtNjIxOTYxOTMyOGJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEyVDA0NTgyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjMDI2NGQ1MjQxNmUxMmI1MjA0MTkyOWRmNjRhMDEzYzUzNTY2OWYwYmQ0NmQyNzllYWY1MjY1YWFiYmQzY2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.AaKLzKXs5osuU2KkAM1Hy7dQhs-ZySYCHE1Cl120kpY)
> 
> I want to know why we need alignment in BEV space, since we already do the alignment in Voxel space. I mean if we align the voxel features in target and source domain，isn't the corresponding BEV features already aligned？or am I missing some details of your method?

This is a good question. We opted to align and transfer knowledge between voxel and BEV features because their representations differ, despite the operation between them involving only pooling. In other words, their representation perspectives differ, which is why we chose to use both.

**Comment by Rayn-Wu** [comment]
Thanks for your reply! Still I have two questions want to check with you.
1、Is the config file "exps/bev_depth_lss_r101_256x704_128x128_24e_ts.py" corresponds to your MATS model? Because I find other config files under “exps” folder，and some of them do not have **discriminator** module. 
2、In https://github.com/flyingGH/BEVUDA-Pytorch the code of the modules and the config files are a little different from this one. Which should I refer to?
